src/NBS.Appointments.Service.Core/Services/QflowService.cs

```diff
@@ -57,12 +59,41 @@ public class QflowService : IQflowService
             var policy = GetRetryPolicy();
             var response = await policy.ExecuteAsync(async (context) => {
                 query["apiSessionId"] = context["SessionId"].ToString();
-                var endpointUrl = QueryHelpers.AddQueryString($"{_options.BaseUrl}/svcCustomAppointment.svc/rest/availability", query);
+                var endpointUrl = QueryHelpers.AddQueryString($"{_options.BaseUrl}/svcCustomAppointment.svc/rest", query);
                 return await client.GetAsync(endpointUrl);
             }, context);
 
             var responseBody = await response.Content.ReadAsStringAsync();
-            return Newtonsoft.Json.JsonConvert.DeserializeObject<SiteAvailabilityResponse[]>(responseBody);
+            return JsonConvert.DeserializeObject<SiteAvailabilityResponse[]>(responseBody);
+        }
+
+        public async Task<SiteSlotAvailabilityByHourResponse> GetSiteSlotAvailabilityAsync(int siteId, DateTime date, string appointmentType)
+        {
+            var query = new Dictionary<string, string>
+            {
+                { "Date", $"{date:yyyy-MM-dd}" },
+                { "SiteId", siteId.ToString() },
+                { "VaccineType", appointmentType },
+                { "ExternalReference", "NotSet" }
+            };
+
+            using var client = _httpClientFactory.CreateClient();
```

Comment:
Lot of repeated code here (same pattern used in method above) - can we refactor?